### ISSUE TITLE
[BUG][JAVA][MICROPROFILE] useRuntimeException is ignored

### ISSUE THREAD

**Issue by ssaip** [OpenAPITools/openapi-generator#14409]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected 
Actual: The generated API throws an org.openapitools.client.**api**.ApiException (which extends Exception)
Expected: org.openapitools.client.ApiException (which extends RuntimeException)

##### Description

When using the java generator with the microprofile library, the `useRuntimeException` flag is ignored.

##### openapi-generator version

openapi-generator-cli-7.0.0-20221012.083708-4.jar

##### OpenAPI declaration file content or url

[sample yaml](https://raw.githubusercontent.com/ssaip/openapi-generator-java-useRuntimeException/main/src/main/openapi/example.yaml)

##### Generation Details

`java -jar openapi-generator-cli-7.0.0-20221012.083708-4.jar generate -g java -i https://raw.githubusercontent.com/ssaip/openapi-generator-java-useRuntimeException/main/src/main/openapi/example.yaml -o C:\temp\ --additional-properties=useRuntimeException=true --library=microprofile`

##### Steps to reproduce

Use the java generator with the microprofile library

##### Related issues/PRs
-

##### Suggest a fix
Either do throw a RuntimeException as configured or (even preferable in my use-case) give the option do NOT generate `ApiException`, `ApiExceptionMapper`, `@RegisterProvider(ApiExceptionMapper.class)` altogether.

If this is not desired for microprofile clients, this information should be added to the documentation.

**Comment by wing328** [comment]
> If this is not desired for microprofile clients, this information should be added to the documentation.

Agreed the doc should be updated on which libraries support this option

May I know if you've time to add `useRuntimeException` support to the java microprofile client?

I can show you some good starting points.

**Comment by ssaip** [comment]
sure, https://github.com/OpenAPITools/openapi-generator/pull/14414

**Comment by Munken** [comment]
@wing328 this can be closed
